packages/vc-verification/src/Ipfs-credential-resolver.ts

```diff
@@ -0,0 +1,77 @@
+import { providers, utils } from 'ethers';
+import { DidStore } from '@ew-did-registry/did-ipfs-store';
+import { IDidStore } from '@ew-did-registry/did-store-interface';
+import { OffChainClaim } from './models';
+import { Resolver } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import * as jwt from 'jsonwebtoken';
+import { upgradeChainId } from './upgrade-chainid';
+import { CredentialResolver } from './credential-resolver';
+
+export class IpfsCredentialResolver implements CredentialResolver {
+  private _ipfsStore: IDidStore;
+  private _resolver: Resolver;
+
+  constructor(
+    provider: providers.Provider,
+    registrySetting: RegistrySettings,
+    didStore: DidStore
+  ) {
+    this._ipfsStore = didStore;
+    this._resolver = new Resolver(provider, registrySetting);
+  }
+
+  /**
+   *
+   * @param serviceEndpoint
+   * @returns {string}
+   */
+  async getCredential(did: string, role: string) {
+    const offChainClaims = await this.offchainClaimsOf(did);
+    return offChainClaims.find(
+      (claim) =>
+        claim.claimType === role || utils.namehash(claim.claimType) === role
+    );
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private async isOffchainClaim(claim: any): Promise<boolean> {
+    const offChainClaimProps = [
+      'claimType',
+      'claimTypeVersion',
+      'issuedToken',
+      'iss',
+    ];
+    const claimProps = Object.keys(claim);
+    return offChainClaimProps.every((p) => claimProps.includes(p));
+  }
+
+  private async offchainClaimsOf(did: string): Promise<OffChainClaim[]> {
+    const transformClaim = (
+      claim: OffChainClaim
+    ): OffChainClaim | undefined => {
+      const transformedClaim: OffChainClaim = { ...claim };
+      return upgradeChainId(transformedClaim);
+    };
+
+    const filterOutMaliciousClaims = (
+      item: OffChainClaim | undefined
+    ): item is OffChainClaim => {
+      return !!item;
+    };
+
+    const didDocument = await this._resolver.read(did);
+    const services = didDocument.service || [];
```

Comment:
services has type `any`, maybe we can add type to it, at least `{serviceEndpoint: string}`

Reply:
It is string by default, since the `services` are of `IServiceEndpoint` type.

Reply:
Okay, then we can do `const services: IServiceEndpoint[] = didDocument.service || [];`. Because in my vscode I see `any` type and this line is marked as error